NEMO/views/timed_services.py

```diff
@@ -409,31 +410,81 @@ def send_email_out_of_time_reservation_notification(request=None):
         if customer.is_staff or customer.is_service_personnel:
             continue
 
-        if area.requires_reservation:
+        threshold = (
+            trigger_time if not area.logout_grace_period else trigger_time - timedelta(minutes=area.logout_grace_period)
+        )
+        physical_access = PhysicalAccessLevel.objects.filter(user=customer, area=area).first()
+        # Check first if allowed schedule has just expired
+        if (
+            physical_access
+            and physical_access.accessible_at(threshold - timedelta(minutes=1))
+            and not physical_access.accessible_at(threshold)
+        ):
+            out_of_time_user_reservations.append(Reservation(user=customer, area=area, end=threshold))
+        else:
+            if area.requires_reservation:
+                ending_reservations = Reservation.objects.filter(
+                    cancelled=False,
+                    missed=False,
+                    shortened=False,
+                    area=area,
+                    user=customer,
+                    start__lte=timezone.now(),
+                    end=threshold,
+                )
+                # find out if a reservation is starting right at the same time (in case of back to back reservations, in which case customer is good)
+                starting_reservations = Reservation.objects.filter(
+                    cancelled=False, missed=False, shortened=False, area=area, user=customer, start=threshold
+                )
+                if ending_reservations.exists() and not starting_reservations.exists():
+                    out_of_time_user_reservations.append(ending_reservations[0])
+
+        # Find all users logged in tools
+        usage_records: List[UsageEvent] = (
```

Comment:
this whole section should be removed. there are no qualification levels or tool requiring reservation in NEMO

Reply:
that was a big overlook, sorry. I fixed it, thanks for spotting.